docs/deployment/gardenlet_api_access.md

```diff
@@ -78,6 +78,10 @@ Extension clients are allowed to perform the same operations as gardenlet client
 - Extension clients are granted the read-only subset of verbs for `CertificateSigningRequests`, `ClusterRoleBindings`, and `ServiceAccounts` (to prevent privilege escalation).
 - Extension clients are granted full access to `Lease` objects but only in the seed-specific namespace.
 
+When the need arises, more exceptions might be added to the access rules for resources that are already handled by the plugins.
+E.g., if an extension needs to populate additional shoot-specific `InternalSecrets`, according handling can be introduced.
```

Comment:
What does this mean? Can you make an example?

Reply:
I just wanted to write down that we don't plan on introducing a generic mechanism for extending permissions for extension clients.

However, I can imagine that we might want to extend permissions for resources that are already handled by the plugins.
For example, right now, the plugins allow the creation of shoot-related `Secrets` and `InternalSecrets` with a hardcoded set of suffixes.
Instead of checking the suffixes, we might allow extensions to create arbitrary `(Internal)Secrets` that have the `<shoot>.` name prefix and handle it accordingly in auth and admission.

Do you get my thoughts?
Should I rephrase this or move it to another place? Or drop it entirely?

Reply:
Got it. No, it's fine this way, just wanted to clarify it with an example :) Thanks!